crengine/src/lvtextfm.cpp

```diff
@@ -1717,7 +1717,7 @@ class LVFormatter {
         LVFont * lastFont = NULL;
         lInt16 lastLetterSpacing = 0;
         int start = 0;
-        int lastWidth = 0;
+        lUInt16 lastWidth = 0;
```

Comment:
I think this one should stay `int`: we use int with `int m_widths`, which is the text formatter buffer and can be > 65535. It's only the local `widths` provided to lvfntman (which works only with lUInt16) that is limited to 65535

Reply:
We have complex spaghetti:
* `static lUInt16 widths[MAX_TEXT_CHUNK_SIZE+1]` in `LVFormatter::measureText()` (static local variable)
* `int * LVFormatter::m_widths`
* `lUInt16 formatted_word_t::width`
* few local variable of the type `int`
* `LVFont::virtual lUInt16 measureText(const lChar32 *text, int len, lUInt16 *widths, lUInt8 *flags, int max_width, lChar32 def_char, TextLangCfg * lang_cfg = NULL, int letter_spacing = 0, bool allow_hyphenation = true, lUInt32 hints = 0, lUInt32 fallbackPassMask = 0)`

And it's all intertwined together. Most likely, I did not fully understand all this. But I mean it is not the size that is important here, but the sign. Sign-reversal integer overflow is painful. I thought I need to get rid of the potential problem. Although not completely fixed :)
Do you really think that `lastWidth` should be of type int?

Reply:
Ok, reverted. There is no desire to study these thousands of lines :)

Reply:
> Do you really think that lastWidth should be of type int?

Yes, int or unsigned int or lInt32 or lUInt32.

> We have complex spaghetti

Not my cooking :) but well, it makes sense to me:

We may do multiple runs of lastFont->measureText() because it's indeed limited to measuring up to what can fit in a lUInt16 (and we provided a max_width=0x7FFF, so half of it, for safety may be, or to avoid signed overflow in case we use some signed local variables without thinking much ?)
Then, after each run, with lastWidth, we always do just:
`lastWidth += widths[len-1]` , that is appending what we measured in the run - and
`m_widths[start + k] = lastWidth + widths[k]` as m_widths can exceed 0xFFFF (fortunately, we can measure paragraphs that are longer than 65535 pixels wide).
Feels like with `unsigned int`, we could measure twice (0xFFFFFFFF) than with `int` (0x7FFFFFFF), but well, they would both be super very long paragraphs, so we should be fine with either.

> lUInt16 formatted_word_t::width

I guess 65535 px is fine for a single word :)